go/vt/vtgate/semantics/analyzer.go

```diff
@@ -455,6 +448,11 @@ func (a *analyzer) noteQuerySignature(node sqlparser.SQLNode) {
 		if node.GroupBy != nil {
 			a.sig.Aggregation = true
 		}
+	case *sqlparser.With:
+		if node.Recursive {
+			// TODO: hacky - we should split this into it's own thing
```

Comment:
What is the hacky thing? I don't follow

Reply:
using aggregation phase rather than it's own CTE phase or something

Reply:
or at least call the signal as recursive cte